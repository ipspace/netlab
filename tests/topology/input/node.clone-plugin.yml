---
plugin: [ node.clone ]

module: [vlan,vrf]

vlans:
 red:
  links: [r-h]  # VLAN links get updated

vrfs:
 red:
  links: [r-h]  # VRF links get updated

groups:
 _auto_create: True   # Test presence of bool flag
 hosts:
  members: [h,host_w_long_name]  # Gets updated with cloned nodes

nodes:
 r:
  device: frr

 h:                   # Short name, regular test
  device: frr
  clone:
   count: 2
   start: 3           # Create h-03 and h-07
   step: 4

 host_w_long_name:    # Test that node names get shortened to at most 16 chars
  device: linux
  clone.count: 3      # Create 3 hosts
  id: 10              # Starting at ID 10

links:
- r-host_w_long_name  # Plugin supports different styles of specifying links, all normalized
- r:
    ifindex: 16       # Custom port on router -> gets incremented
  h:
    mtu: 1600         # Interface atts get copied too
