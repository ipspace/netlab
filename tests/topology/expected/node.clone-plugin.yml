groups:
  hosts:
    members:
    - h-03
    - h-07
    - host_w_long_n-01
    - host_w_long_n-02
    - host_w_long_n-03
input:
- topology/input/node.clone-plugin.yml
- package:topology-defaults.yml
lag:
  lacp: fast
  lacp_mode: active
  mode: 802.3ad
links:
- _linkname: links[1]
  interfaces:
  - ifindex: 16
    ifname: eth16
    ipv4: 10.1.0.2/30
    node: r
  - ifindex: 1
    ifname: eth1
    ipv4: 10.1.0.1/30
    mtu: 1600
    node: h-03
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: p2p
- _linkname: links[2]
  interfaces:
  - ifindex: 17
    ifname: eth17
    ipv4: 10.1.0.6/30
    node: r
  - ifindex: 1
    ifname: eth1
    ipv4: 10.1.0.5/30
    mtu: 1600
    node: h-07
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 10.1.0.4/30
  type: p2p
- _linkname: links[3]
  bridge: input_3
  interfaces:
  - ifindex: 30000
    ifname: bond5
    ipv4: 10.1.0.9/30
    node: r
  - ifindex: 30000
    ifname: bond5
    ipv4: 10.1.0.10/30
    node: h-03
  lag:
    ifindex: 5
  linkindex: 3
  node_count: 2
  prefix:
    ipv4: 10.1.0.8/30
  type: lag
- _linkname: links[4]
  bridge: input_4
  interfaces:
  - ifindex: 30001
    ifname: bond6
    ipv4: 10.1.0.13/30
    node: r
  - ifindex: 30000
    ifname: bond6
    ipv4: 10.1.0.14/30
    node: h-07
  lag:
    ifindex: 6
  linkindex: 4
  node_count: 2
  prefix:
    ipv4: 10.1.0.12/30
  type: lag
- _linkname: links[5]
  bridge: input_5
  interfaces:
  - ifindex: 30002
    ifname: bond1
    ipv4: 10.1.0.17/30
    node: r
  - ifindex: 30001
    ifname: bond1
    ipv4: 10.1.0.18/30
    node: h-03
  lag:
    ifindex: 1
  linkindex: 5
  node_count: 2
  prefix:
    ipv4: 10.1.0.16/30
  type: lag
- _linkname: links[6]
  bridge: input_6
  interfaces:
  - ifindex: 30003
    ifname: bond2
    ipv4: 10.1.0.21/30
    node: r
  - ifindex: 30001
    ifname: bond2
    ipv4: 10.1.0.22/30
    node: h-07
  lag:
    ifindex: 2
  linkindex: 6
  node_count: 2
  prefix:
    ipv4: 10.1.0.20/30
  type: lag
- _linkname: links[7]
  bridge: input_7
  gateway:
    ipv4: 172.16.1.1/24
  interfaces:
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.1.1/24
    node: r
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.1.10/24
    node: host_w_long_n-01
  linkindex: 7
  node_count: 2
  prefix:
    ipv4: 172.16.1.0/24
  role: stub
  type: lan
- _linkname: links[8]
  bridge: input_8
  gateway:
    ipv4: 172.16.2.1/24
  interfaces:
  - ifindex: 2
    ifname: eth2
    ipv4: 172.16.2.1/24
    node: r
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.2.11/24
    node: host_w_long_n-02
  linkindex: 8
  node_count: 2
  prefix:
    ipv4: 172.16.2.0/24
  role: stub
  type: lan
- _linkname: links[9]
  bridge: input_9
  gateway:
    ipv4: 172.16.3.1/24
  interfaces:
  - ifindex: 3
    ifname: eth3
    ipv4: 172.16.3.1/24
    node: r
  - ifindex: 1
    ifname: eth1
    ipv4: 172.16.3.12/24
    node: host_w_long_n-03
  linkindex: 9
  node_count: 2
  prefix:
    ipv4: 172.16.3.0/24
  role: stub
  type: lan
- _linkname: links[3].lag[1]
  interfaces:
  - ifindex: 4
    ifname: eth4
    node: r
  - ifindex: 2
    ifname: eth2
    node: h-03
  lag:
    _parentindex: 3
  linkindex: 10
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[3].lag[2]
  interfaces:
  - ifindex: 5
    ifname: eth5
    node: r
  - ifindex: 3
    ifname: eth3
    node: h-03
  lag:
    _parentindex: 3
  linkindex: 11
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[4].lag[1]
  interfaces:
  - ifindex: 6
    ifname: eth6
    node: r
  - ifindex: 2
    ifname: eth2
    node: h-07
  lag:
    _parentindex: 4
  linkindex: 12
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[4].lag[2]
  interfaces:
  - ifindex: 7
    ifname: eth7
    node: r
  - ifindex: 3
    ifname: eth3
    node: h-07
  lag:
    _parentindex: 4
  linkindex: 13
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[5].lag[1]
  interfaces:
  - ifindex: 20
    ifname: eth20
    node: r
  - ifindex: 4
    ifname: eth4
    node: h-03
  lag:
    _parentindex: 5
  linkindex: 14
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[5].lag[2]
  interfaces:
  - ifindex: 21
    ifname: eth21
    node: r
  - ifindex: 5
    ifname: eth5
    node: h-03
  lag:
    _parentindex: 5
  linkindex: 15
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[6].lag[1]
  interfaces:
  - ifindex: 22
    ifname: eth22
    node: r
  - ifindex: 4
    ifname: eth4
    node: h-07
  lag:
    _parentindex: 6
  linkindex: 16
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[6].lag[2]
  interfaces:
  - ifindex: 23
    ifname: eth23
    node: r
  - ifindex: 5
    ifname: eth5
    node: h-07
  lag:
    _parentindex: 6
  linkindex: 17
  node_count: 2
  prefix: false
  type: p2p
- _linkname: vlans.red.links[1]
  bridge: input_18
  interfaces:
  - _vlan_mode: irb
    ifindex: 8
    ifname: eth8
    ipv4: 172.16.0.1/24
    node: r
    vlan:
      access: red
  - _vlan_mode: irb
    ifindex: 6
    ifname: eth6
    ipv4: 172.16.0.2/24
    node: h-03
    vlan:
      access: red
  linkindex: 18
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
  vlan:
    access: red
- _linkname: vlans.red.links[2]
  bridge: input_19
  interfaces:
  - _vlan_mode: irb
    ifindex: 9
    ifname: eth9
    ipv4: 172.16.0.1/24
    node: r
    vlan:
      access: red
  - _vlan_mode: irb
    ifindex: 6
    ifname: eth6
    ipv4: 172.16.0.3/24
    node: h-07
    vlan:
      access: red
  linkindex: 19
  node_count: 2
  prefix:
    allocation: id_based
    ipv4: 172.16.0.0/24
  type: lan
  vlan:
    access: red
- _linkname: vrfs.red.links[1]
  interfaces:
  - ifindex: 10
    ifname: eth10
    ipv4: 10.1.0.26/30
    node: r
  - ifindex: 7
    ifname: eth7
    ipv4: 10.1.0.25/30
    node: h-03
  linkindex: 20
  node_count: 2
  prefix:
    ipv4: 10.1.0.24/30
  type: p2p
  vrf: red
- _linkname: vrfs.red.links[2]
  interfaces:
  - ifindex: 11
    ifname: eth11
    ipv4: 10.1.0.30/30
    node: r
  - ifindex: 7
    ifname: eth7
    ipv4: 10.1.0.29/30
    node: h-07
  linkindex: 21
  node_count: 2
  prefix:
    ipv4: 10.1.0.28/30
  type: p2p
  vrf: red
module:
- lag
- vlan
- vrf
name: input
nodes:
  h-03:
    af:
      ipv4: true
      vpnv4: true
    box: debian/bookworm64
    device: frr
    id: 2
    interfaces:
    - ifindex: 1
      ifname: eth1
      ipv4: 10.1.0.1/30
      linkindex: 1
      mtu: 1600
      name: h-03 -> r
      neighbors:
      - ifname: eth16
        ipv4: 10.1.0.2/30
        node: r
      type: p2p
    - ifindex: 30000
      ifname: bond5
      ipv4: 10.1.0.10/30
      lag:
        ifindex: 5
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 3
      name: h-03 -> r
      neighbors:
      - ifname: bond5
        ipv4: 10.1.0.9/30
        node: r
      type: lag
      virtual_interface: true
    - ifindex: 30001
      ifname: bond1
      ipv4: 10.1.0.18/30
      lag:
        ifindex: 1
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 5
      name: h-03 -> r
      neighbors:
      - ifname: bond1
        ipv4: 10.1.0.17/30
        node: r
      type: lag
      virtual_interface: true
    - ifindex: 2
      ifname: eth2
      lag:
        _parentindex: 3
      linkindex: 10
      name: h-03 -> r
      neighbors:
      - ifname: eth4
        node: r
      type: p2p
    - ifindex: 3
      ifname: eth3
      lag:
        _parentindex: 3
      linkindex: 11
      name: h-03 -> r
      neighbors:
      - ifname: eth5
        node: r
      type: p2p
    - ifindex: 4
      ifname: eth4
      lag:
        _parentindex: 5
      linkindex: 14
      name: h-03 -> r
      neighbors:
      - ifname: eth20
        node: r
      type: p2p
    - ifindex: 5
      ifname: eth5
      lag:
        _parentindex: 5
      linkindex: 15
      name: h-03 -> r
      neighbors:
      - ifname: eth21
        node: r
      type: p2p
    - bridge: input_18
      ifindex: 6
      ifname: eth6
      linkindex: 18
      name: '[Access VLAN red] h-03 -> r'
      neighbors:
      - ifname: eth8
        ipv4: 172.16.0.1/24
        node: r
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 7
      ifname: eth7
      ipv4: 10.1.0.25/30
      linkindex: 20
      name: h-03 -> r
      neighbors:
      - ifname: eth10
        ipv4: 10.1.0.26/30
        node: r
        vrf: red
      type: p2p
      vrf: red
    - bridge_group: 1
      ifindex: 8
      ifname: vlan1000
      ipv4: 172.16.0.2/24
      name: VLAN red (1000) -> [r,h-07]
      neighbors:
      - ifname: vlan1000
        ipv4: 172.16.0.1/24
        node: r
      - ifname: vlan1000
        ipv4: 172.16.0.3/24
        node: h-07
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: red
    lag:
      lacp: fast
      lacp_mode: active
      mode: 802.3ad
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.2/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module:
    - lag
    - vlan
    - vrf
    name: h-03
    vlan:
      max_bridge_group: 1
    vlans:
      red:
        bridge_group: 1
        id: 1000
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
    vrf:
      as: 65000
    vrfs:
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  h-07:
    af:
      ipv4: true
      vpnv4: true
    box: debian/bookworm64
    device: frr
    id: 3
    interfaces:
    - ifindex: 1
      ifname: eth1
      ipv4: 10.1.0.5/30
      linkindex: 2
      mtu: 1600
      name: h-07 -> r
      neighbors:
      - ifname: eth17
        ipv4: 10.1.0.6/30
        node: r
      type: p2p
    - ifindex: 30000
      ifname: bond6
      ipv4: 10.1.0.14/30
      lag:
        ifindex: 6
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 4
      name: h-07 -> r
      neighbors:
      - ifname: bond6
        ipv4: 10.1.0.13/30
        node: r
      type: lag
      virtual_interface: true
    - ifindex: 30001
      ifname: bond2
      ipv4: 10.1.0.22/30
      lag:
        ifindex: 2
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 6
      name: h-07 -> r
      neighbors:
      - ifname: bond2
        ipv4: 10.1.0.21/30
        node: r
      type: lag
      virtual_interface: true
    - ifindex: 2
      ifname: eth2
      lag:
        _parentindex: 4
      linkindex: 12
      name: h-07 -> r
      neighbors:
      - ifname: eth6
        node: r
      type: p2p
    - ifindex: 3
      ifname: eth3
      lag:
        _parentindex: 4
      linkindex: 13
      name: h-07 -> r
      neighbors:
      - ifname: eth7
        node: r
      type: p2p
    - ifindex: 4
      ifname: eth4
      lag:
        _parentindex: 6
      linkindex: 16
      name: h-07 -> r
      neighbors:
      - ifname: eth22
        node: r
      type: p2p
    - ifindex: 5
      ifname: eth5
      lag:
        _parentindex: 6
      linkindex: 17
      name: h-07 -> r
      neighbors:
      - ifname: eth23
        node: r
      type: p2p
    - bridge: input_19
      ifindex: 6
      ifname: eth6
      linkindex: 19
      name: '[Access VLAN red] h-07 -> r'
      neighbors:
      - ifname: eth9
        ipv4: 172.16.0.1/24
        node: r
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 7
      ifname: eth7
      ipv4: 10.1.0.29/30
      linkindex: 21
      name: h-07 -> r
      neighbors:
      - ifname: eth11
        ipv4: 10.1.0.30/30
        node: r
        vrf: red
      type: p2p
      vrf: red
    - bridge_group: 1
      ifindex: 8
      ifname: vlan1000
      ipv4: 172.16.0.3/24
      name: VLAN red (1000) -> [h-03,r]
      neighbors:
      - ifname: vlan1000
        ipv4: 172.16.0.2/24
        node: h-03
      - ifname: vlan1000
        ipv4: 172.16.0.1/24
        node: r
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: red
    lag:
      lacp: fast
      lacp_mode: active
      mode: 802.3ad
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.3/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    module:
    - lag
    - vlan
    - vrf
    name: h-07
    vlan:
      max_bridge_group: 1
    vlans:
      red:
        bridge_group: 1
        id: 1000
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
    vrf:
      as: 65000
    vrfs:
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
  host_w_long_n-01:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 10
    interfaces:
    - bridge: input_7
      gateway:
        ipv4: 172.16.1.1/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.1.10/24
      linkindex: 7
      name: host_w_long_n-01 -> r
      neighbors:
      - ifname: eth1
        ipv4: 172.16.1.1/24
        node: r
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.110
      mac: 08:4f:a9:00:00:0a
    name: host_w_long_n-01
    role: host
  host_w_long_n-02:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 11
    interfaces:
    - bridge: input_8
      gateway:
        ipv4: 172.16.2.1/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.2.11/24
      linkindex: 8
      name: host_w_long_n-02 -> r
      neighbors:
      - ifname: eth2
        ipv4: 172.16.2.1/24
        node: r
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.111
      mac: 08:4f:a9:00:00:0b
    name: host_w_long_n-02
    role: host
  host_w_long_n-03:
    af:
      ipv4: true
    box: generic/ubuntu2004
    device: linux
    id: 12
    interfaces:
    - bridge: input_9
      gateway:
        ipv4: 172.16.3.1/24
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.3.12/24
      linkindex: 9
      name: host_w_long_n-03 -> r
      neighbors:
      - ifname: eth3
        ipv4: 172.16.3.1/24
        node: r
      role: stub
      type: lan
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.112
      mac: 08:4f:a9:00:00:0c
    name: host_w_long_n-03
    role: host
  r:
    _set_ifindex: true
    af:
      ipv4: true
      vpnv4: true
    box: debian/bookworm64
    device: frr
    id: 1
    interfaces:
    - bridge: input_7
      ifindex: 1
      ifname: eth1
      ipv4: 172.16.1.1/24
      linkindex: 7
      name: r -> host_w_long_n-01
      neighbors:
      - ifname: eth1
        ipv4: 172.16.1.10/24
        node: host_w_long_n-01
      role: stub
      type: lan
    - bridge: input_8
      ifindex: 2
      ifname: eth2
      ipv4: 172.16.2.1/24
      linkindex: 8
      name: r -> host_w_long_n-02
      neighbors:
      - ifname: eth1
        ipv4: 172.16.2.11/24
        node: host_w_long_n-02
      role: stub
      type: lan
    - bridge: input_9
      ifindex: 3
      ifname: eth3
      ipv4: 172.16.3.1/24
      linkindex: 9
      name: r -> host_w_long_n-03
      neighbors:
      - ifname: eth1
        ipv4: 172.16.3.12/24
        node: host_w_long_n-03
      role: stub
      type: lan
    - ifindex: 4
      ifname: eth4
      lag:
        _parentindex: 3
      linkindex: 10
      name: r -> h-03
      neighbors:
      - ifname: eth2
        node: h-03
      type: p2p
    - ifindex: 5
      ifname: eth5
      lag:
        _parentindex: 3
      linkindex: 11
      name: r -> h-03
      neighbors:
      - ifname: eth3
        node: h-03
      type: p2p
    - ifindex: 6
      ifname: eth6
      lag:
        _parentindex: 4
      linkindex: 12
      name: r -> h-07
      neighbors:
      - ifname: eth2
        node: h-07
      type: p2p
    - ifindex: 7
      ifname: eth7
      lag:
        _parentindex: 4
      linkindex: 13
      name: r -> h-07
      neighbors:
      - ifname: eth3
        node: h-07
      type: p2p
    - bridge: input_18
      ifindex: 8
      ifname: eth8
      linkindex: 18
      name: '[Access VLAN red] r -> h-03'
      neighbors:
      - ifname: eth6
        ipv4: 172.16.0.2/24
        node: h-03
      type: lan
      vlan:
        access: red
        access_id: 1000
    - bridge: input_19
      ifindex: 9
      ifname: eth9
      linkindex: 19
      name: '[Access VLAN red] r -> h-07'
      neighbors:
      - ifname: eth6
        ipv4: 172.16.0.3/24
        node: h-07
      type: lan
      vlan:
        access: red
        access_id: 1000
    - ifindex: 10
      ifname: eth10
      ipv4: 10.1.0.26/30
      linkindex: 20
      name: r -> h-03
      neighbors:
      - ifname: eth7
        ipv4: 10.1.0.25/30
        node: h-03
        vrf: red
      type: p2p
      vrf: red
    - ifindex: 11
      ifname: eth11
      ipv4: 10.1.0.30/30
      linkindex: 21
      name: r -> h-07
      neighbors:
      - ifname: eth7
        ipv4: 10.1.0.29/30
        node: h-07
        vrf: red
      type: p2p
      vrf: red
    - bridge_group: 1
      ifindex: 12
      ifname: vlan1000
      ipv4: 172.16.0.1/24
      name: VLAN red (1000) -> [h-03,h-07]
      neighbors:
      - ifname: vlan1000
        ipv4: 172.16.0.2/24
        node: h-03
      - ifname: vlan1000
        ipv4: 172.16.0.3/24
        node: h-07
      type: svi
      virtual_interface: true
      vlan:
        mode: irb
        name: red
    - ifindex: 13
      ifname: eth13
      linkindex: 0
      neighbors: []
      remote_id: 1
      remote_ifindex: 666
      type: p2p
    - ifindex: 14
      ifname: eth14
      linkindex: 0
      neighbors: []
      remote_id: 1
      remote_ifindex: 666
      type: p2p
    - ifindex: 15
      ifname: eth15
      linkindex: 0
      neighbors: []
      remote_id: 1
      remote_ifindex: 666
      type: p2p
    - ifindex: 16
      ifname: eth16
      ipv4: 10.1.0.2/30
      linkindex: 1
      name: r -> h-03
      neighbors:
      - ifname: eth1
        ipv4: 10.1.0.1/30
        node: h-03
      type: p2p
    - ifindex: 17
      ifname: eth17
      ipv4: 10.1.0.6/30
      linkindex: 2
      name: r -> h-07
      neighbors:
      - ifname: eth1
        ipv4: 10.1.0.5/30
        node: h-07
      type: p2p
    - ifindex: 18
      ifname: eth18
      linkindex: 0
      neighbors: []
      remote_id: 1
      remote_ifindex: 666
      type: p2p
    - ifindex: 19
      ifname: eth19
      linkindex: 0
      neighbors: []
      remote_id: 1
      remote_ifindex: 666
      type: p2p
    - ifindex: 20
      ifname: eth20
      lag:
        _parentindex: 5
      linkindex: 14
      name: r -> h-03
      neighbors:
      - ifname: eth4
        node: h-03
      type: p2p
    - ifindex: 21
      ifname: eth21
      lag:
        _parentindex: 5
      linkindex: 15
      name: r -> h-03
      neighbors:
      - ifname: eth5
        node: h-03
      type: p2p
    - ifindex: 22
      ifname: eth22
      lag:
        _parentindex: 6
      linkindex: 16
      name: r -> h-07
      neighbors:
      - ifname: eth4
        node: h-07
      type: p2p
    - ifindex: 23
      ifname: eth23
      lag:
        _parentindex: 6
      linkindex: 17
      name: r -> h-07
      neighbors:
      - ifname: eth5
        node: h-07
      type: p2p
    - ifindex: 30000
      ifname: bond5
      ipv4: 10.1.0.9/30
      lag:
        ifindex: 5
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 3
      name: r -> h-03
      neighbors:
      - ifname: bond5
        ipv4: 10.1.0.10/30
        node: h-03
      type: lag
      virtual_interface: true
    - ifindex: 30001
      ifname: bond6
      ipv4: 10.1.0.13/30
      lag:
        ifindex: 6
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 4
      name: r -> h-07
      neighbors:
      - ifname: bond6
        ipv4: 10.1.0.14/30
        node: h-07
      type: lag
      virtual_interface: true
    - ifindex: 30002
      ifname: bond1
      ipv4: 10.1.0.17/30
      lag:
        ifindex: 1
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 5
      name: r -> h-03
      neighbors:
      - ifname: bond1
        ipv4: 10.1.0.18/30
        node: h-03
      type: lag
      virtual_interface: true
    - ifindex: 30003
      ifname: bond2
      ipv4: 10.1.0.21/30
      lag:
        ifindex: 2
        lacp: fast
        lacp_mode: active
        mode: 802.3ad
      linkindex: 6
      name: r -> h-07
      neighbors:
      - ifname: bond2
        ipv4: 10.1.0.22/30
        node: h-07
      type: lag
      virtual_interface: true
    lag:
      lacp: fast
      lacp_mode: active
      mode: 802.3ad
    libvirt:
      nic_adapter_count: 24
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.1/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module:
    - lag
    - vlan
    - vrf
    name: r
    vlan:
      max_bridge_group: 1
    vlans:
      red:
        bridge_group: 1
        id: 1000
        mode: irb
        prefix:
          allocation: id_based
          ipv4: 172.16.0.0/24
    vrf:
      as: 65000
    vrfs:
      red:
        af:
          ipv4: true
        export:
        - '65000:1'
        id: 1
        import:
        - '65000:1'
        rd: '65000:1'
        vrfidx: 100
plugin:
- node.clone
provider: libvirt
vlans:
  red:
    host_count: 0
    id: 1000
    neighbors:
    - ifname: vlan1000
      ipv4: 172.16.0.2/24
      node: h-03
    - ifname: vlan1000
      ipv4: 172.16.0.1/24
      node: r
    - ifname: vlan1000
      ipv4: 172.16.0.3/24
      node: h-07
    prefix:
      allocation: id_based
      ipv4: 172.16.0.0/24
vrf:
  as: 65000
vrfs:
  red:
    export:
    - '65000:1'
    id: 1
    import:
    - '65000:1'
    rd: '65000:1'
